test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c

```diff
@@ -124,998 +103,149 @@ void test_vProcessGeneratedUDPPacket_CantSendPacket( void )
-
-    usGenerateProtocolChecksum_ExpectAndReturn( pucLocalEthernetBuffer, xLocalNetworkBuffer.xDataLength, pdTRUE, 0 );
-
-    xLocalNetworkBuffer.pxEndPoint = &xEndPoint;
-    xLocalNetworkBuffer.pxEndPoint->pxNetworkInterface = &xInterface;
-    xLocalNetworkBuffer.pxEndPoint->pxNetworkInterface->pfOutput = &NetworkInterfaceOutputFunction_Stub;
+    uint16_t usPort = 0x1234;
+    BaseType_t xIsWaitingForARPResolution = 0;
 
-    vProcessGeneratedUDPPacket( &xLocalNetworkBuffer );
-
-    TEST_ASSERT_EQUAL( xLocalNetworkBuffer.usPort, pxUDPPacket->xUDPHeader.usDestinationPort );
-    TEST_ASSERT_EQUAL( ipPROTOCOL_ICMP, pxUDPPacket->xIPHeader.ucProtocol );
-    TEST_ASSERT_EQUAL( 0, pxUDPPacket->xUDPHeader.usChecksum );
+    catch_assert( xProcessReceivedUDPPacket( NULL, usPort, &xIsWaitingForARPResolution ) );
 }
 
-/*
- * @brief Test the asserts in the function.
+/**
+ * @brief Input with null buffer pointer in network buffer.
  */
-void test_xProcessReceivedUDPPacket_catchAsserts( void )
+void test_xProcessReceivedUDPPacket_NullBuffer( void )
 {
-    NetworkBufferDescriptor_t xLocalNetworkBuffer;
-    BaseType_t xIsWaitingARPResolution = pdFALSE;
-
-    catch_assert( xProcessReceivedUDPPacket( NULL, 0, &xIsWaitingARPResolution ) );
-
-    xLocalNetworkBuffer.pucEthernetBuffer = NULL;
-    catch_assert( xProcessReceivedUDPPacket( &xLocalNetworkBuffer, 0, &xIsWaitingARPResolution ) );
-}
-
-/*
- * @brief Test when there is no listening socket and the packet is not
- *        for this node.
- */
-void test_xProcessReceivedUDPPacket_NoListeningSocket_NotForThisNode( void )
-{
-    NetworkBufferDescriptor_t xLocalNetworkBuffer;
-    uint16_t usPort = 0;
-    uint8_t pucLocalEthernetBuffer[ ipconfigTCP_MSS ];
-    BaseType_t xResult;
-    BaseType_t xIsWaitingARPResolution = pdFALSE;
-    UDPPacket_t * pxUDPPacket;
+    uint16_t usPort = 0x1234;
+    BaseType_t xIsWaitingForARPResolution = 0;
+    NetworkBufferDescriptor_t xNetworkBuffer;
 
-    /* Cleanup the ethernet buffer. */
-    memset( pucLocalEthernetBuffer, 0, ipconfigTCP_MSS );
-
-    xLocalNetworkBuffer.pucEthernetBuffer = pucLocalEthernetBuffer;
-    pxUDPPacket = ( ( const UDPPacket_t * ) xLocalNetworkBuffer.pucEthernetBuffer );
-    pxUDPPacket->xEthernetHeader.usFrameType = ipIPv4_FRAME_TYPE;
-
-    /* No socket found. */
-    pxUDPSocketLookup_ExpectAndReturn( usPort, NULL );
+    memset( &xNetworkBuffer, 0, sizeof( xNetworkBuffer ) );
+    xNetworkBuffer.pucEthernetBuffer = NULL;
 
-    xResult = xProcessReceivedUDPPacket( &xLocalNetworkBuffer, usPort, &xIsWaitingARPResolution );
-    TEST_ASSERT_EQUAL( pdFAIL, xResult );
-    TEST_ASSERT_EQUAL( pdFALSE, xIsWaitingARPResolution );
+    catch_assert( xProcessReceivedUDPPacket( &xNetworkBuffer, usPort, &xIsWaitingForARPResolution ) );
 }
 
-/*
- * @brief Test when there is no listening socket but the packet seems like
- *        a late DNS response.
+/**
+ * @brief Pass IPv6 packet to IPv4 API to handle.
```

Comment:
Comment should be : Pass IPv4 packet to IPv4 API to handle.

Reply:
Updated